libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/recenter/RecenterButtonComponent.kt

```diff
@@ -1,40 +1,50 @@
 package com.mapbox.navigation.dropin.component.recenter
 
-import android.view.View.GONE
-import android.view.View.VISIBLE
+import androidx.core.view.isVisible
 import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
 import com.mapbox.navigation.core.MapboxNavigation
 import com.mapbox.navigation.dropin.component.camera.CameraAction
 import com.mapbox.navigation.dropin.component.camera.CameraViewModel
 import com.mapbox.navigation.dropin.component.camera.TargetCameraMode
+import com.mapbox.navigation.dropin.component.navigation.NavigationStateViewModel
+import com.mapbox.navigation.dropin.component.navigationstate.NavigationState
 import com.mapbox.navigation.dropin.lifecycle.UIComponent
 import com.mapbox.navigation.dropin.view.MapboxExtendableButton
 import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.launch
 
 @ExperimentalPreviewMapboxNavigationAPI
 internal class RecenterButtonComponent(
     private val cameraViewModel: CameraViewModel,
+    private val navigationStateViewModel: NavigationStateViewModel,
     private val recenterButton: MapboxExtendableButton,
 ) : UIComponent() {
 
     override fun onAttached(mapboxNavigation: MapboxNavigation) {
         super.onAttached(mapboxNavigation)
 
-        coroutineScope.launch {
-            cameraViewModel.state.collect {
-                if (it.cameraMode == TargetCameraMode.Idle) {
-                    recenterButton.visibility = VISIBLE
-                } else {
-                    recenterButton.visibility = GONE
-                }
-            }
-        }
         recenterButton.setOnClickListener {
             cameraViewModel.invoke(
                 CameraAction.ToFollowing
             )
         }
+
+        coroutineScope.launch {
+            combine(
+                cameraViewModel.state,
+                navigationStateViewModel.state
+            ) { cameraState, navigationState ->
+                when (navigationState) {
+                    is NavigationState.RoutePreview -> {
+                        recenterButton.isVisible = false
+                    }
+                    else -> {
+                        recenterButton.isVisible = cameraState.cameraMode == TargetCameraMode.Idle
+                    }
+                }
+            }.collect()
+        }
```

Comment:
```suggestion
        coroutineScope.launch {
            combine(
                cameraViewModel.state,
                navigationStateViewModel.state
            ) { cameraState, navigationState ->
                navigationState == NavigationState.RoutePreview || cameraState.cameraMode == TargetCameraMode.Idle
                }
            }.collect { isVisible ->
                recenterButton.isVisible = isVisible
            }
        }
```

Reply:
boolean logic on my refactor may be wrong. But the suggestion here is to make the "collect" block update the recenterButton visibility

Reply:
Fixed.